### PR TITLE
.NET 5.0

### LINKED ISSUE #2078: long type can not convert implicitly to int type for dotnet 5.0
**Describe the bug**
update the framework of neo project to net5.0, and reference neo-vm master branch. Build failed.

**To Reproduce**
1. Open the neo-cli and neo project, update the framework to net5.0
2. Build the project.

**Expected behavior**
Build successfully.

**Screenshots**
1f687c704dc061673e783c56999ef67

**Platform:**
 - OS: Windows 10 x64
 - Version master

### PR DESCRIPTION
**Pull request by erikzhang**
Close #2078

### CODE REVIEW

**Comment by devhawk** [comment]
Are we expecting to merge this before preview 4?

**Review by shargon — APPROVED** [review]

**Comment by erikzhang** [comment]
Merge?

**Comment by shargon** [comment]
> Are we expecting to merge this before preview 4?

I think that it's good, we can test it more. What do you think?

**Comment by erikzhang** [comment]
@neo-project/ngd-shanghai

**Comment by devhawk** [comment]
> > Are we expecting to merge this before preview 4?
> 
> I think that it's good, we can test it more. What do you think?

If you're asking me @shargon, I think we should get preview 4 out the door first.

**Comment by cloud8little** [comment]
under test

**Comment by superboyiii** [comment]
> @neo-project/ngd-shanghai
I think it should be merged after Preview4.

File: src/neo/neo.csproj [hunk 1]
```diff
@@ -21,14 +21,14 @@
...
     <PackageReference Include="BouncyCastle.NetCore" Version="1.8.8" />
-    <PackageReference Include="K4os.Compression.LZ4" Version="1.1.11" />
+    <PackageReference Include="K4os.Compression.LZ4" Version="1.2.6" />
     <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel" Version="2.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.WebSockets" Version="2.2.1" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="3.1.3" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="3.1.3" />
-    <PackageReference Include="Neo.VM" Version="3.0.0-CI00251" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="5.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="5.0.0" />
```
Comment (cloud8little) [inline src/neo/neo.csproj:30]: version 5.0.0 of Microsoft.EntityFrameworkCore.Sqlite and Microsoft.Extensions.Configuration.Json are incompatible with RpcServer plugin.

**Comment by cloud8little** [comment]
When Microsoft.EntityFrameworkCore.Sqlite and Microsoft.Extensions.Configuration.Json are both version "3.1.3", smoke test passed.

When Microsoft.EntityFrameworkCore.Sqlite or Microsoft.Extensions.Configuration.Json is version "5.0.0", Rpc call failed here:
https://github.com/neo-project/neo-modules/blob/master/src/RpcServer/RpcServer.cs#L199

**Comment by chenzhitong** [comment]
> When Microsoft.EntityFrameworkCore.Sqlite and Microsoft.Extensions.Configuration.Json are both version "3.1.3", smoke test passed.
> 
> When Microsoft.EntityFrameworkCore.Sqlite or Microsoft.Extensions.Configuration.Json is version "5.0.0", Rpc call failed here:
> https://github.com/neo-project/neo-modules/blob/master/src/RpcServer/RpcServer.cs#L199

https://github.com/neo-project/neo-modules/pull/407

**Comment by chenzhitong** [comment]
https://github.com/neo-project/neo/blob/master/.github/workflows/main.yml#L24-L25

Maybe it should be updated to

```yml
dotnet tool install --version 5.0.142902 --tool-path ./ dotnet-format --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json
./dotnet-format . --check -v diagnostic
```

**Comment by cloud8little** [comment]
1d3a2b5aa5ede7aacbe4054fd1373d7

6119a9ddbbd7946897b0a8d3efc1544

Exception Stack Info:

```
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=Microsoft.AspNetCore.Server.Kestrel.Core
  StackTrace:
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.CreateResponseHeader(Boolean appCompleted)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProduceStart(Boolean appCompleted)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.InitializeResponseAsync(Int32 firstWriteByteCount)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.WriteAsync(ReadOnlyMemory`1 data, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseStream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Http.HttpResponseWritingExtensions.WriteAsync(HttpResponse response, String text, Encoding encoding, CancellationToken cancellationToken)
   at Neo.Plugins.RpcServer.<ProcessAsync>d__22.MoveNext() in D:\Github\workflow\neo-modules\src\RpcServer\RpcServer.cs:line 199
```

**Comment by cloud8little** [comment]
> > When Microsoft.EntityFrameworkCore.Sqlite and Microsoft.Extensions.Configuration.Json are both version "3.1.3", smoke test passed.
> > When Microsoft.EntityFrameworkCore.Sqlite or Microsoft.Extensions.Configuration.Json is version "5.0.0", Rpc call failed here:
> > https://github.com/neo-project/neo-modules/blob/master/src/RpcServer/RpcServer.cs#L199
> 
> [neo-project/neo-modules#407](https://github.com/neo-project/neo-modules/pull/407)

No, it is not.

File: src/neo/neo.csproj (hunk 1 above)
Comment (vncoelho) [reply src/neo/neo.csproj:30]: This is not the first time we have this incompatibilities with `Microsoft.EntityFrameworkCore.Sqlite`

**Comment by erikzhang** [comment]
Now it can be merged.

**Comment by devhawk** [comment]
> Now it can be merged.

@erikzhang are we waiting until after preview 4?

**Comment by erikzhang** [comment]
Preview4 nuget package has been released.

**Review by shargon — APPROVED** [review]

**Pull request merged**